extensions/integration-tests/src/schemaCompare.test.ts

```diff
@@ -0,0 +1,184 @@
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+'use strict';
+
+import 'mocha';
+import * as azdata from 'azdata';
+import * as utils from './utils';
+const path = require('path');
+import { context } from './testContext';
+import assert = require('assert');
+import { getStandaloneServer } from './testConfig';
+import { readFileSync, unlinkSync, existsSync } from 'fs';
+
+let schemaCompareService: azdata.SchemaCompareServicesProvider;
+let dacpac1: string = path.join(__dirname, 'testData/Database1.dacpac');
+let dacpac2: string = path.join(__dirname, 'testData/Database2.dacpac');
+let dummyDBName: string = 'ads_schemaCompareDB'; // This is used as fill in name and not created anywhere
+
+if (context.RunTest) {
+	suite('Schema compare integration test suite', () => {
+		suiteSetup(async function () {
+			let attemps: number = 20;
+			while (attemps > 0) {
+				schemaCompareService = await azdata.dataprotocol.getProvider<azdata.SchemaCompareServicesProvider>('MSSQL', azdata.DataProviderType.SchemaCompareServicesProvider);
+				if (schemaCompareService) {
+					break;
+				}
+				attemps--;
+				await utils.sleep(1000); // To ensure the providers are registered.
+			}
+			console.log(`Start schema compare tests`);
+		});
+		test('Schema compare dacpac to dacpac comparision and script generation', async function () {
+			assert(schemaCompareService, 'Schema Compare Service Provider is not available');
+
+			let source: azdata.SchemaCompareEndpointInfo = {
+				endpointType: azdata.SchemaCompareEndpointType.dacpac,
+				packageFilePath: dacpac1,
+				databaseName: '',
+				ownerUri: '',
+			};
+			let target: azdata.SchemaCompareEndpointInfo = {
+				endpointType: azdata.SchemaCompareEndpointType.dacpac,
+				packageFilePath: dacpac2,
+				databaseName: '',
+				ownerUri: '',
+			};
+
+			let schemaCompareResult = await schemaCompareService.schemaCompare(source, target, azdata.TaskExecutionMode.execute);
+			assertSchemaCompareResult(schemaCompareResult);
+		});
+
+		test('Schema compare database to database comparision and script generation', async function () {
+
+			let server = await getStandaloneServer();
+			await utils.connectToServer(server, 300);
+
+			let nodes = <azdata.objectexplorer.ObjectExplorerNode[]>await azdata.objectexplorer.getActiveConnectionNodes();
+			assert(nodes.length > 0, `Expecting at least one active connection, actual: ${nodes.length}`);
+
+			let index = nodes.findIndex(node => node.nodePath.includes(server.serverName));
+			assert(index !== -1, `Failed to find server: "${server.serverName}" in OE tree`);
+
+			let ownerUri = await azdata.connection.getUriForConnection(nodes[index].connectionId);
+			let now = new Date();
+
+			let sourceDB: string = 'ads_schemaCompare_sourceDB_' + now.getTime().toString();
```

Comment:
I'd add the client machine to the name - you can see who is leaking database and will prevent against name collisions (we had this issue with the DacFx tests running concurrently on 5 clients)